cmd/internal/find/primary/meta/stringPredicate.go

```diff
@@ -0,0 +1,29 @@
+package meta
+
+import (
+	"fmt"
+
+	"github.com/puppetlabs/wash/cmd/internal/find/primary/errz"
+)
+
+// StringPredicate => [^-]+
```

Comment:
I think regex/glob support should be next, since this isn't really that useful right now. I'm thinking the user would type something like
```
find instances .Tags[] r'termination_date'
```
where the regex grammar is `r<regex>`. Note that `r'termination_date'` would be passed as `rtermination_date` into the parser (`'`/`"` is syntactic sugar).

Reply:
We may need to distinguish string literals by requiring users to double quote them. Or we could preserve the existing behavior of "fallback to literal string".

Reply:
Also, on my Iterm shell, we don't need to quite `.Tags[]`. It works as-is.

Reply:
On the fence here. I like making the regex behavior easy (I think `ag` went that way). But that can also be annoying. I guess I'm comfortable saying we'll need some sigil to denote a regex, unless we think people will almost always use a regex instead of an exact string match.

Reply:
Yeah, I think people using a regex is more likely than an exact string match so we should make it easy. Guess that would mean `StringPredicate`'s semantics would be matching the regex compiled by the token. If exact string matches are common, and those matches are likely to contain regexp characters, we could always add quoted string literals to the grammar.

If you like ^, I can go ahead and add it to the PR. Can do it separately too since it's big enough.

Reply:
Hmm, maybe we do need a sigil to avoid weird magic. For example, if someone types `[abc]`, that will be parsed as an ArrayPredicate and a syntax error will be thrown. I could make ArrayPredicate return match errors so it can fall-back to StringPredicate, but that seems pretty hacky and can be hard to reason about if `meta` gets more complicated.

I guess telling people "Hey, you need to type r'<regex>'" for regex support isn't that bad.